onnx_tf/backend.py

```diff
@@ -267,43 +267,57 @@ def supports_device(cls, device):
 
   @classmethod
   def onnx_graph_to_tensorflow_ops(cls,
-                                   graph_def,
+                                   subgraph,
                                    input_values,
+                                   tensor_dict,
                                    opset=None,
                                    strict=True):
     """
     Converts ONNX graph to Tensorflow operations
     Args:
-      graph_def:        the ONNX graph to be converted
+      subgraph:         the ONNX graph to be converted
       input_values:     dictionary with values/tensors to initialize
-                        the graph inputs. the dictionary must contain values
-                        for all the graph_def.input
+                        the subgraph inputs. if the subgraph.input
+                        are send in as parameters then it is required,
+                        otherwise this can be empty dictionary
+      tensor_dict:      the dictionary that contain values for all the
+                        node.inputs in the subgraph that are not defined
+                        in the subgraph or input_values.
       opset:            opset version of the operator set.
       strict:           whether to enforce semantic equivalence between the
                         original model and the converted tensorflow model,
                         defaults to True (yes, enforce semantic equivalence).
     Returns:
       array of Tensorflow Tensors
     """
-    input_dict_items = []
-    # set input values for the subgraph
-    for value_info in graph_def.input:
-      if value_info.name in input_values:
-        x = input_values[value_info.name]
-        input_dict_items.append((value_info.name, x))
-
-    tensor_dict = dict(input_dict_items)
-
-    for node in graph_def.node:
+    # get the subgraph.input from input_values
+    subgraph_tensor_dict = dict(input_values)
```

Comment:
Is 'dict(...)' needed? I believe input_values is already a dictionary.

Reply:
@chinhuang007 I want to create a new dictionary call subgraph_tensor_dict with the value in input_values. And the change I made in subgraph_tensor_dict later in the code will not modify any values in input_values

Reply:
If you want to create a copy of dictionary, I would suggest use deepcopy() instead of dict(...) which is shallow copy.